src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/FileMoveNotificationListener.cs

```diff
@@ -136,10 +138,7 @@ async Task<List<(Renamer.RenameDocumentActionSet, string)>> GetNamespaceUpdateAc
                     }
 
                     // This is a file item to another directory, it should only detect this a Update Namespace action.
-                    // TODO Upgrade this api to get rid of the exclamation sign
-#pragma warning disable CS0618 // Type or member is obsolete https://github.com/dotnet/project-system/issues/8591
-                    Renamer.RenameDocumentActionSet documentAction = await Renamer.RenameDocumentAsync(oldDocument, null!, documentFolders);
-#pragma warning restore CS0618 // Type or member is obsolete
+                    Renamer.RenameDocumentActionSet documentAction = await Renamer.RenameDocumentAsync(oldDocument, s_renameOptions, null, documentFolders);
```

Comment:
Similarly I would expect this to be:

```suggestion
                    DocumentRenameOptions options = new(
                        RenameMatchingTypeInStrings: optionSet.GetOption(RenameOptions.RenameInStrings),
                        RenameMatchingTypeInComments: optionSet.GetOption(RenameOptions.RenameInComments));
                
                    Renamer.RenameDocumentActionSet documentAction = await Renamer.RenameDocumentAsync(oldDocument, options, newDocumentName: null, documentFolders);
```

Reply:
The exact same code to do this exists in Roslyn but is also marked obsolete... I'm trying to understand why now.
https://github.com/dotnet/roslyn/blob/4d0af36500e90653f5086866acccf2a73567aaa0/src/Workspaces/Core/Portable/Rename/Renamer.cs#L27-L39

```C#
        [Obsolete]
        private static SymbolRenameOptions GetSymbolRenameOptions(OptionSet optionSet)
            => new(
                RenameOverloads: optionSet.GetOption(RenameOptions.RenameOverloads),
                RenameInStrings: optionSet.GetOption(RenameOptions.RenameInStrings),
                RenameInComments: optionSet.GetOption(RenameOptions.RenameInComments),
                RenameFile: false);


        [Obsolete]
        private static DocumentRenameOptions GetDocumentRenameOptions(OptionSet optionSet)
            => new(
                RenameMatchingTypeInStrings: optionSet.GetOption(RenameOptions.RenameInStrings),
                RenameMatchingTypeInComments: optionSet.GetOption(RenameOptions.RenameInComments));
```

Reply:
They cannot, as it currently stands. We could add UI for this, as called out above. Would those options default to the solution's setting values? Would changing them also change the solution's settings?

Reply:
The options don't have values associated with a solution. If there is no way to set these in the Project System UI then use anything setting that makes the most sense for this feature.

Inline Rename feature sets these values from its UI and remembers those settings across VS sessions.

![image](https://user-images.githubusercontent.com/41759/213943114-ffab9f8c-491b-40a5-a9bd-b33bae614034.png)

I don't think these should necessarily be used by project system though.

Reply:
> The options don't have values associated with a solution

This snippet of the obsolete code suggests there is an option accessible via the solution object, even if that's actually at the machine or user level, for example:

```c#
solution.Options.GetOption(RenameOptions.RenameOverloads)
```

Reply:
`Solution.Options` is gonna be obsolete soon as well.

Reply:
@tmat where will the IDE store these settings then? Perhaps we should read them to behave consistent with user preferences.

We could use `Solution.Options` for now and cross that bridge when we get there.

Reply:
The settings do not exist. The only settings that exist are the ones that Inline Rename UI uses. Should the settings used by the Project System when renaming document be the same as the latest value specified in Inline Rename UI?

Reply:
In lieu of creating new UI or linking to existing UI/settings, for the purposes of simply removing deprecated API calls, would using a defaulted `DocumentRenameOptions`/`SymbolRenameOptions` be adequate to maintain current functionality? If we want to improve the experience, I'm down for that. However, we should make that into a new issue/feature and prioritize it with our other work.

Reply:
> would using a defaulted `DocumentRenameOptions`/`SymbolRenameOptions` be adequate to maintain current functionality? 

Yup. That should do.

Reply:
@drewnoakes Do you think it is okay if we use the defaults for these types (`DocumentRenameOptions`/`SymbolRenameOptions`) for now, and make an issue to add a feature for exposing these options to users?

Reply:
> > would using a defaulted `DocumentRenameOptions`/`SymbolRenameOptions` be adequate to maintain current functionality?
> 
> Yup. That should do.

@tmat Right now the way we're using the deprecated API means that Roslyn will pick up the `OptionSet` from `document.Project.Solution.Options`. Are you saying that will currently _always_ be the same as the options we get by doing `new DocumentRenameOptions()`?

Reply:
Yes.